src/main/scala/scala/compat/java8/converterImpl/MakesSteppers.scala

```diff
@@ -0,0 +1,160 @@
+trait MakesDoubleSeqStepper extends Any {
+  /** Generates a fresh stepper over `Double`s suitable for sequential use */
+  def stepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Double` steppers for map keys suitable for sequential use. */
+trait MakesDoubleKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Double` suitable for sequential use */
+  def keyStepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Double` steppers  for map values suitable for sequential use. */
+trait MakesDoubleValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Double` suitable for sequential use */
+  def valueStepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers suitable for sequential use. */
+trait MakesIntSeqStepper extends Any {
+  /** Generates a fresh stepper over `Int`s suitable for sequential use */
+  def stepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers for map keys suitable for sequential use. */
+trait MakesIntKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Int` suitable for sequential use */
+  def keyStepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers for map values suitable for sequential use. */
+trait MakesIntValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Int` suitable for sequential use */
+  def valueStepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers suitable for sequential use. */
+trait MakesLongSeqStepper extends Any {
+  /** Generates a fresh stepper over `Long`s suitable for sequential use */
+  def stepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers for map keys suitable for sequential use. */
+trait MakesLongKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Long` suitable for sequential use */
+  def keyStepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers for map values suitable for sequential use. */
+trait MakesLongValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Long` suitable for sequential use */
+  def valueStepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create generic steppers suitable for sequential or parallel use. */
+trait MakesAnyStepper[A] extends Any {
+  /** Generates a fresh stepper over `A`s that can be efficiently subdivided */
+  def stepper: AnyStepper[A] with EfficientSubstep
+}
+
+/** Classes or objects implementing this trait create generic steppers for map keys suitable for sequential or parallel use. */
+trait MakesAnyKeyStepper[A] extends Any {
```

Comment:
Thinking further: Even if you want it prevent inefficient steppers statically, you could ask for a function that returns the right stepper instead of taking an object that implements one of the `Make...` traits. That would also allow abstracting over key / value / sequence steppers.